test/test_evaluators.py

```diff
@@ -187,38 +187,46 @@ class TestMultilabelClassificationEvaluator(unittest.TestCase):
     TARGETS = np.array([[1, 0, 0],
                         [0, 1, 1],
                         [1, 1, 1]])
-    PREDICTIONS = np.array([[1, 0.3, 0],
-                            [0, 1, 0.5],
-                            [0.5, 0.6, 0.5]])
+    PROB_PREDICTIONS = np.array([[1, 0.3, 0],
+                                [0, 1, 0.5],
+                                [0.5, 0.6, 0.5]])
+    INDEX_PREDICTIONS = np.array([[0, 1, 2],
+                                 [1, 2, 0],
+                                 [1, 0, 2]])
 
     def test_precision_evaluator(self):
         thresholds = [0.0, 0.3, 0.6, 0.7]
         expectations = [0.66666, 0.83333, 1.0, 0.66666]
         for i in range(len(thresholds)):
             prec_eval = PrecisionEvaluator(ThresholdPredictionFilter(thresholds[i]))
-            prec_eval.add_predictions(self.PREDICTIONS, self.TARGETS)
+            prec_eval.add_predictions(self.PROB_PREDICTIONS, self.TARGETS)
             self.assertAlmostEqual(prec_eval.get_report(average='samples')[f"precision_thres={thresholds[i]}"], expectations[i], places=4)
 
         ks = [1, 2, 3]
         expectations = [1.0, 0.833333, 0.66666]
         for i in range(len(ks)):
             prec_eval = PrecisionEvaluator(TopKPredictionFilter(ks[i]))
-            prec_eval.add_predictions(self.PREDICTIONS, self.TARGETS)
+            prec_eval.add_predictions(self.PROB_PREDICTIONS, self.TARGETS)
             self.assertAlmostEqual(prec_eval.get_report(average='samples')[f"precision_top{ks[i]}"], expectations[i], places=4)
 
     def test_recall_evaluator(self):
         thresholds = [0.0, 0.3, 0.6, 0.7]
         expectations = [1.0, 1.0, 0.61111, 0.5]
         for i in range(len(thresholds)):
             recall_eval = RecallEvaluator(ThresholdPredictionFilter(thresholds[i]))
-            recall_eval.add_predictions(self.PREDICTIONS, self.TARGETS)
+            recall_eval.add_predictions(self.PROB_PREDICTIONS, self.TARGETS)
             self.assertAlmostEqual(recall_eval.get_report(average='samples')[f"recall_thres={thresholds[i]}"], expectations[i], places=4)
 
         ks = [0, 1, 2, 3]
         expectations = [0, 0.61111, 0.88888, 1.0]
         for i in range(len(ks)):
             recall_eval = RecallEvaluator(TopKPredictionFilter(ks[i]))
-            recall_eval.add_predictions(self.PREDICTIONS, self.TARGETS)
+            recall_eval.add_predictions(self.PROB_PREDICTIONS, self.TARGETS)
+            self.assertAlmostEqual(recall_eval.get_report(average='samples')[f"recall_top{ks[i]}"], expectations[i], places=4)
+
+        for i in range(len(ks)):
+            recall_eval = RecallEvaluator(TopKPredictionFilter(ks[i], prediction_mode='indices'))
```

Comment:
here is the unit test for TopKPredictionFilter when `prediction_mode` is `indices` @pj-ms